pkg/prometheus/operator.go

```diff
@@ -310,6 +311,16 @@ func New(ctx context.Context, conf operator.Config, logger log.Logger, r prometh
 		c.nsPromInf = newNamespaceInformer(c, c.config.Namespaces.PrometheusAllowList)
 	}
 
+	isSupportEndpointSliceResource := false
+	_, err = c.kclient.DiscoveryV1beta1().EndpointSlices(kubeletObjectNamespace).List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
```

Comment:
Is this intended to automatically detect whether the endpointslice resource is present in the cluster?

Reply:
yes